internal/services/network/virtual_network_gateway_connection_resource.go

```diff
@@ -186,6 +186,26 @@ func resourceVirtualNetworkGatewayConnection() *pluginsdk.Resource {
 				},
 			},
 
+			"custom_bgp_addresses": {
+				Type:     pluginsdk.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"primary_address": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.Any(validation.IsIPv4Address),
+						},
+						"secondary_address": {
```

Comment:
as address is in the containing block:
```suggestion
						"primary": {
							Type:         pluginsdk.TypeString,
							Required:     true,
							ValidateFunc: validation.Any(validation.IsIPv4Address),
						},
						"secondary": {
```

Reply:
I changed all the properties like the suggestion